lib/daemon_runner/semaphore.rb

```diff
@@ -15,9 +15,15 @@ class << self
       #
       def lock(name, limit = 3, **options)
         options.merge!(name: name)
-        lock = Semaphore.new(options)
-        lock.lock
-        lock
+        semaphore = Semaphore.new(options)
+        semaphore.lock
+        if block_given?
+          lock_thr = semaphore.renew
+          yield
+          lock_thr.kill
```

Comment:
In the `Thread` there is a [`loop do`](https://github.com/rapid7/daemon_runner/blob/master/lib/daemon_runner/semaphore.rb#L74-L79) so it's never going to return.